### DISCUSSION TITLE
Support for Snowflake

### DISCUSSION THREAD

**Discussion by ialluri** [diesel-rs/diesel#2863] [Ideas]
Hi,

I have a diesel-rust application that works for Postgres.  I need to change the datastore from Postgres to Snowflake. I was wondering if there is an initiative that is in progress to support Snowflake, or if any assessment was conducted if this is even feasible.

~ialluri

**Comment by weiznich** [comment]
Diesel currently supports backends for Postgresql, Sqlite and Mysql out of the box. We do not plan to add additional backends any time soon. That written the main crate is designed in such a way that it is possible to develop custom backends as third party crates. I'm not aware of any crate that supports Snowflake. We are happy to answer questions that come up while developing such a backend. Existing examples for such crates are [diesel-oci](https://github.com/GiGainfosystems/diesel-oci) and [rsfbclient](https://github.com/fernandobatels/rsfbclient). See https://github.com/diesel-rs/diesel/discussions/2824 for similar discussion.
